sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/AppConfigurationReplicaClientsBuilder.java

```diff
@@ -182,8 +192,38 @@ List<AppConfigurationReplicaClient> buildClients(ConfigStore configStore) {
         return clients;
     }
 
-    private AppConfigurationReplicaClient modifyAndBuildClient(ConfigurationClientBuilder builder, String endpoint, Integer replicaCount) {
-        TracingInfo tracingInfo = new TracingInfo(isDev, isKeyVaultConfigured, replicaCount, Configuration.getGlobalConfiguration());
+    public AppConfigurationReplicaClient buildClient(String failoverEndpoint, ConfigStore configStore) {
+
+        if (StringUtils.hasText(configStore.getConnectionString())) {
+            ConnectionString connectionString = new ConnectionString(configStore.getConnectionString());
+            connectionString.setUri(failoverEndpoint);
+            ConfigurationClientBuilder builder = createBuilderInstance().connectionString(connectionString.toString());
+            return modifyAndBuildClient(builder, failoverEndpoint, 0);
+        } else if (configStore.getConnectionStrings().size() > 0) {
+            ConnectionString connectionString = new ConnectionString(configStore.getConnectionStrings().get(0));
+            connectionString.setUri(failoverEndpoint);
+            ConfigurationClientBuilder builder = createBuilderInstance().connectionString(connectionString.toString());
+            return modifyAndBuildClient(builder, failoverEndpoint, 0);
+        } else {
+            ConfigurationClientBuilder builder = createBuilderInstance();
+            if (!credentialConfigured) {
+                // System Assigned Identity. Needs to be checked last as all of the above should
+                // have an Endpoint.
+                LOGGER.debug(
+                    "Connecting to {} using Azure System Assigned Identity or Azure User Assigned Identity.",
```

Comment:
Same here

Reply:
System, I think this comment/log is from the old 2.x version. Updated.